src/control/server/mgmt_svc.go

```diff
@@ -532,3 +532,34 @@ func (svc *mgmtSvc) ListPools(ctx context.Context, req *mgmtpb.ListPoolsReq) (*m
 
 	return resp, nil
 }
+
+func (svc *mgmtSvc) LeaderQuery(ctx context.Context, req *mgmtpb.LeaderQueryReq) (*mgmtpb.LeaderQueryResp, error) {
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+
+	if len(svc.harness.Instances()) == 0 {
+		return nil, errors.New("no I/O servers configured; can't determine leader")
+	}
+
+	instance := svc.harness.Instances()[0]
+	sb := instance.getSuperblock()
+	if sb == nil {
+		return nil, errors.New("no I/O superblock found; can't determine leader")
+	}
+
+	if req.System != sb.System {
+		return nil, errors.Errorf("received leader query for wrong system (local: %q, req: %q)",
+			sb.System, req.System)
+	}
+
+	leaderAddr, err := instance.msClient.LeaderAddress()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to determine current leader address")
+	}
```

Comment:
Wondering if some of this logic should be in some shared GetLeader function. Many of these MgmtSvc hooks get the leader instance before doing anything else.

Reply:
I thought about that but decided it would be better to do later, as for 1.0 we're only going to support a single MS instance, and didn't want to complicate that code before we needed to.

Reply:
Works for me.